migration/idempotent/017-metric-rollup.sql

```diff
@@ -0,0 +1,282 @@
+-- create_metric_rollup prepares the given rollup. The actual rollup views are created by _prom_catalog.scan_for_new_rollups()
+CREATE OR REPLACE PROCEDURE _prom_catalog.create_metric_rollup(name TEXT, resolution INTERVAL, retention INTERVAL) AS
+$$
+    DECLARE
+        schema_name TEXT := 'ps_' || name;
+        exists BOOLEAN;
+
+    BEGIN
+        EXECUTE FORMAT('SELECT count(*) > 0 FROM _prom_catalog.rollup WHERE name = %L', name) INTO exists;
+        IF exists THEN
+            RAISE EXCEPTION 'ERROR: cannot create metric rollup for %. REASON: already exists.', name;
+        END IF;
+        -- We do not use IF NOT EXISTS (below) since we want to stop creation process if the schema already exists.
+        -- This forces the user to delete the conflicting schema before proceeding ahead.
+        EXECUTE FORMAT('CREATE SCHEMA %s', schema_name);
+        INSERT INTO _prom_catalog.rollup VALUES (name, schema_name, resolution, retention);
+    END;
+$$
+LANGUAGE PLPGSQL;
+
+-- delete_metric_rollup deletes everything related to the given metric rollup label name. It does the following:
+-- 1. Delete the entries in _prom_catalog.rollup
+-- 2. Delete the entries in _prom_catalog.metric_with_rollup
+-- 3. Delete the actual Cagg views that represent the rollups. This is done by directing deleting the entire rollup schema
+CREATE OR REPLACE PROCEDURE _prom_catalog.delete_metric_rollup(rollup_name TEXT) AS
+$$
+    DECLARE
+        rollup_schema_name TEXT;
+
+    BEGIN
+        EXECUTE FORMAT('SELECT schema_name FROM _prom_catalog.rollup WHERE name = %L', rollup_name) INTO rollup_schema_name;
+        IF rollup_schema_name IS NULL THEN
+            RAISE EXCEPTION '[Rollup] Cannot delete metric rollup due to rollup_schema_name being NULL';
```

Comment:
This message might be a bit confusing when read outside of the context of this function (i.e. in a log file).

Maybe something like this is better:
`Cannot delete metric rollup [rollup name], missing schema [schema name]`